sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -107,8 +108,7 @@ export class BatchingReceiver extends MessageReceiver {
             this._receiver.removeListener(ReceiverEvents.receiverDrained, onReceiveDrain);
           }
           this.isReceivingMessages = false;
-          const err = translate(this.detachedError);
-          return reject(err);
+          return resolve([]);
```

Comment:
Since we are no longer throwing an error, our previous discussion around PeekLock vs ReceiveAndDelete comes into play again.

We should check the ReceiveMode of the receiver and return the collected messages if we are in ReceiveAndDelete mode.

Reply:
Ok I understand but lets say we call `receiveBatch(300)` with `ReceiveAndDelete` mode and we add 300 credits on the receiver link. Service bus adds 300 messages at once on the link and delete all the messages(because we are receiving in ReceiveAndDelete mode) and we receive one by one. if network goes down lets say after receiving 200 messages then we lost 100 messages.

I also think we should return whatever messages we received(in this case 200) to the user, but just wanted to check in with you, is that fine to loose 100 messages?